azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/AzureStorageCheckpointLeaseManager.java

```diff
@@ -36,9 +36,49 @@
     private String storageContainerName;
     private String storageBlobPrefix;
     
-    private CloudBlobClient storageClient;
-    private CloudBlobContainer eventHubContainer;
-    private CloudBlobDirectory consumerGroupDirectory;
+    private class ProtectedStorageClient
```

Comment:
>ProtectedStorageClient [](start = 18, length = 22)

This approach lookslike is a workaround to the real underlying storage issue and is not the right thing to do :(

 After doing some research on storage client code-base, here's one plausible approach to take, please consider this:
- options.setMaximumExecutionTimeInMs(AzureStorageCheckpointLeaseManager.storageMaximumExecutionTimeInMs) <-- This setting dictates "when" the TimeoutException is being thrown with this string "The client could not finish the operation within specified maximum execution timeout" (this one is literally a client error)
- the real transient error thrown by the Service is never surfacing up and here's why (ExecutionEngine.java - has the retrypolicy implementation on every storage client op):
   - the retryLogic of Storage Client is: 
         if (retrycount >= maxRetryCount)
                  throw realTransientException;
         else if (clientOperationTime > maxExecutionTime)
                  throw StorageException(timeout);
        .......

so, I guess, we need to tune our MaxRetryCount to hit first - then the maxExecutionTime.